tests/unit/test_migration_engine.py

```diff
@@ -0,0 +1,197 @@
+from typing import Any, Dict
+
+from pytest import raises
+
+from nxdrive.dao.migrations.migration import MigrationInterface
+from nxdrive.dao.migrations.migration_engine import MigrationEngine
+
+
+def generate_migration(
+    schema_version: int, previous_schema_version: int, /, *, should_raise: bool = False
+):
+    """Instantiate and return a migration object."""
+
+    class TestMigration(MigrationInterface):
+        table_name = "".join(random.choice(string.ascii_letters) for _ in range(5))
+
+        def __init__(self, schema_version, previous_schema_version) -> None:
+            self._schema_version = schema_version
+            self._previous_schema_version = previous_schema_version
+
+        def upgrade(self, cursor: Cursor) -> None:
+            """Create a table with a random name or generate an exception."""
+            if not should_raise:
+                cursor.execute(
+                    f"CREATE TABLE {self.table_name} ("
+                    "   path STRING NOT NULL,"
+                    "   PRIMARY KEY (path)"
+                    ")"
+                )
+                cursor.execute(f"PRAGMA user_version = {self.version}")
+            else:
+                raise sqlite3.Error("Mocked error")
+
+        def downgrade(self, cursor: Cursor) -> None:
+            """Drop the created table or generate an exception."""
+            if not should_raise:
+                cursor.execute(f"DROP TABLE {self.table_name}")
+                cursor.execute(f"PRAGMA user_version = {self.previous_version}")
+            else:
+                raise sqlite3.Error("Mocked error")
+
+        @property
+        def version(self) -> int:
+            return self._schema_version
+
+        @property
+        def previous_version(self) -> int:
+            return self._previous_schema_version
+
+    return TestMigration(schema_version, previous_schema_version)
+
+
+def generate_migrations_dict(count: int) -> Dict[str, Any]:
+    """Generate a dictionary of migration objects."""
+    migrations_dict = {}
+    previous_migration = 0
+
+    for x in range(1, count):
+        migration = generate_migration(x, previous_migration)
+        migrations_dict.update({f"{x}_added_table_{migration.table_name}": migration})
+        previous_migration = x
+
+    return migrations_dict
+
+
+def test_migration_engine_no_error():
+    """A simple test to see if the downgrade remove all the upgrade modifications."""
+    migrations_dict = generate_migrations_dict(10)
+
+    def old_migrations_callback(*_, **__):
+        pass
+
+    with sqlite3.connect(":memory:") as conn:
+        cursor = conn.cursor()
+        migration_engine = MigrationEngine(conn, migrations_dict)
+
+        # Let's do a database upgrade
+        migration_engine.execute_database_upgrade(0, 0, old_migrations_callback)
+        user_version = cursor.execute("PRAGMA user_version").fetchone()[0]
+        assert user_version == 9
+
+        # Let's do a database downgarde to version 5
```

Comment:
```suggestion
        # Let's do a database downgrade to version 5
```

Reply:
it's strange that codespell didn't see it